x/evm/embeds/contracts/IOracle.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.19;
+
+/// @dev Oracle interface for querying exchange rates
+interface IOracle {
+  /// @dev queryExchangeRate queries the exchange rate for a given pair
+  /// @param pair the pair to query
+  function queryExchangeRate(string memory pair) external returns (string memory);
```

Comment:
```suggestion
  function queryExchangeRate(string memory pair) external view returns (string memory);
```

We should add the `view` keyword here since the contract doesn't modify state.